libdnf/transaction/transaction_item_action.cpp

```diff
@@ -50,20 +50,20 @@ std::string transaction_item_action_to_letter(TransactionItemAction action) {
     switch (action) {
         case TransactionItemAction::INSTALL:
             return "I";
-        case TransactionItemAction::DOWNGRADE:
-            return "D";
         case TransactionItemAction::UPGRADE:
             return "U";
+        case TransactionItemAction::DOWNGRADE:
+            return "D";
+        case TransactionItemAction::REINSTALL:
+            return "R";
         case TransactionItemAction::REMOVE:
             // "R" is for Reinstall, therefore use "E" for rEmove (or Erase)
             return "E";
-        case TransactionItemAction::REINSTALL:
-            return "R";
+        case TransactionItemAction::REPLACED:
+            return "O";  // TODO(lukash) historically Obsolete, do we change this?
```

Comment:
We can change it, but what would be a better letter for the operation?

Reply:
Yeah, I don't know, either the first free letter, which would be "P" (I don't like that as I find just confusing), or something explicitly different like e.g. "X"?

Reply:
`O` is not wrong because all reverse action can be understand as an obsolete, but explicit or implicit.

Reply:
Yes, but `O` is now not obviously related to `Replaced`. I think it will be confusing on the UI. Seeing `Replaced` as the long version, people will not know why its short version is `O` (figuring out it's for `O`bsolete is a medium difficulty puzzle, and we shouldn't subject our users to puzzles :slightly_smiling_face: ). I find `X` somewhat better in this regard, though not by that much...

Reply:
Also, `O` understood as Obsolete would be quite misleading, because obsolete is specifically when a package is replaced by a different one, which will not be the case here 99% of the time.